crates/providers/Cargo.toml

```diff
@@ -11,6 +11,7 @@ opt-level = "z"
 lto = true
 
 [workspace]
+resolver = "2"
```

Comment:
Could you help me understand why this is necessary?

Reply:
This was added due to a warning, likely because of the use of `wasmtime` -- that's gone now so this isn't necessary.

Reply:
NOTE: the warning is *still* there without the `wasmtime` dependency:

```
warning: some crates are on edition 2021 which defaults to `resolver = "2"`, but virtual workspaces default to `resolver = "1"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
```

I get the warning on `main`

Reply:
I feel like this change is out of scope for this PR if this already happens on `main`, could we leave it out for now?

Reply:
It's a warning, and it should be fixed -- this code fixes it, I'd like to keep it in.

If it's emitting a warning on `main` it should be fixed -- it's the equivalent of a lint in my mind, not sure why we would leave out fixing it.

Reply:
Thank you for explaining your point, Victor.
I've never seen this warning on `main` and I work with these crates almost every working day. We've never seen these warnings in CI and nobody else has reported them.
Clearly, there is something different in your environment and first we should figure out why exactly this happens and *how* to fix it and not just blindly add stopgap solutions. All crates in the workspace should be using the same Rust edition, if they do not - that is a bug.
Please file an issue with clear reproduction steps including your `cargo` version and we will discuss how to proceed there, this change is out-of-scope for provider bindgen, and the diff in this PR is already huge. Please try to focus on making atomic, incremental changes.

This change may potentially affect things like MSRV, for example, among other things, so we have to give it more thought first to evaluate all the options and tradeoffs

Reply:
>  I've never seen this warning on main and I work with these crates almost every working day. We've never seen these warnings in CI and nobody else has reported them.

I don't see *any* warnings from the build process in CI -- in fact I don't see any output from `cargo` in CI builds is because it's 99% `nix` output. I'm not convinced we'd *know* if warnings were happening.

> Clearly, there is something different in your environment and first we should figure out why exactly this happens and how to fix it and not just blindly add stopgap solutions. All crates in the workspace should be using the same Rust edition, if they do not - that is a bug.

This isn't a stop-gap solution -- it's very clear what it does. 

The fact that it's a bug is precisely why this should be fixed -- see the last line:

```
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
```

The intended resolver for the code-base is 2021 -- if I'm getting the warning, that means it's *not* set to 2021 for some crates (and I get the same error on `main`, so it's not the code in this PR) -- allowing code in that's not set to the right resolver would be a bug.

> This change may potentially affect things like MSRV, for example, among other things, so we have to give it more thought first to evaluate all the options and tradeoffs

Could you explain your specific concern around this a bit more? We do not set a `rust-version` in the crates -- there *is no* MSRV currently for `wasmcloud/wasmcloud`.  Edition `2021` *defaults* to resolver 2, which is exactly what I'm ensuring with this change.

Reply:
After chatting, I'll file an issue about this so someone can kick off an investigation